src/Microsoft.AspNet.Mvc.ViewFeatures/ViewComponents/DefaultViewComponentHelper.cs

```diff
@@ -237,7 +249,23 @@ private ViewComponentDescriptor SelectComponent(Type componentType)
                     Resources.FormatViewComponent_IViewComponentFactory_ReturnedNull(descriptor.Type.FullName));
             }
 
+            WriteTelemetryIfEnabled("Microsoft.AspNet.Mvc.BeforeViewComponent", context);
+
             invoker.Invoke(context);
+
+            WriteTelemetryIfEnabled("Microsoft.AspNet.Mvc.AfterViewComponent", context);
+        }
+
+        private void WriteTelemetryIfEnabled(string telemetryName, ViewComponentContext context)
+        {
+#pragma warning disable 0618
+            if (_telemetry.IsEnabled(telemetryName))
+            {
+                _telemetry.WriteTelemetry(
+                    telemetryName,
+                    new { viewComponentContext = context });
```

Comment:
It looks like it's included as a member of the VCC so we're good.

Let's add the actiondescriptor as its own top level item here